batch-job.go

```diff
@@ -37,16 +38,36 @@ const (
 	BatchJobReplicate BatchJobType = "replicate"
 	BatchJobKeyRotate BatchJobType = "keyrotate"
 	BatchJobExpire    BatchJobType = "expire"
+	BatchJobCatalog   BatchJobType = "catalog"
 )
 
 // SupportedJobTypes supported job types
+//
+// Deprecated: Use ClientSupportedJobTypes instead.
```

Comment:
What is the point of deprecating this? 

I guess it also comes down to that I don't understand what "client" refers to in `ClientSupportedJobTypes`.

Reply:
Just that it is ambiguously named. With this change, catalog job is only going to be supported by EOS and not open-source minio. So `ClientSupportedJobTypes` is going to be used only in mc and other client applications.